cudapoa/src/cudapoa_kernels.cuh

```diff
@@ -216,11 +216,11 @@ typedef struct GraphDetails
  */
 
 void generatePOA(claragenomics::cudapoa::OutputDetails* output_details_d,
```

Comment:
hmm I see that we're having to use `void` type for the structs since our implementation is in the `.cu` file, and if we make this templated then the whole thing would have to be included in the `.cuh`. do you want to give that a shot? Main reason I'm suggesting that is we won't have to cast types in `cudapoa_kernels.cu` if we make this function templated as well.

```
template<typename SizeT, typename ScoreT>
generatePOA(claragenomics::cudapoa::OutputDetails* output_details_d,
                      claragenomics::cudapoa::InputDetails<SizeT>* Input_details_d,
                      ...
)
```

what do you think?

Reply:
same for the other functions in this

Reply:
actually initially I had implemented it that way, but the code structurally got damaged, all those codes from `cudapoa_add_alignment.cu`, `cudapoa_nw.cu`,  `cudapoa_generate_consensus.cu`, `cudapoa_topsort` had to be moved under the same file in `cudapoa_kernels.cuh` leaving those files blank. I don't remember if I also faced circular dependency or not, but already did not like how the code looked like.  That's why I rewrote it back this way.

Reply:
Oh hmm... does it not work if each of those files are `#include <cudapoa_generate_consensus.cuh>` into the `cudapoa_kernels.cuh` file? so keeping the in different files but just including them into a single unit?

Reply:
sorry, my comment was not clear enough. There are multiple functions in `cuda_kernels.cuh` with void pointers.  Namely: `generatePOA()`, `addAlignment()`, `runNW()`, `generateConsensusTestHost()` and `runTopSort()`. Other places where any of these functions are called, include `cuda_kernels.cuh`.
Now by making these functions template type, we need to either bring function definition (and all their dependent functions) into the same `cuda_kernels.cuh` file which as mentioned in my previous comment it won't have a nice structure. If we want to keep dependencies in their own file, e.g. we define  `addAlignment()` inside `cuda_kernels.cuh` but keep all the other functions defined in `cudapoa_add_alignments.cu` (i.e `addAlignmentKernel()` and `addAlignmentToGraph()` ) where they are, then there will be circular dependency, i.e. in `cudapoa_add_alignment.cu` we have to include 'cudapoa_kernels.cuh' and vice-versa.

or alternatively we could move each of these function declarations out of `cuda_kernels.cuh` and keep it where it's being defined. Then there will be 2 problems: 1- caller may require to decide what template type to use (something like what is done inside `generatePOA()` by calling `use32bitScore()`) 2- we don't have a single .cuh file to keep all POA kernels under the same umbrella.